packages/core/src/external/aws/lambda-logic/bundle-to-html.ts

```diff
@@ -586,25 +590,102 @@ function createDiagnosticReportsSection(
     false
   );
 
-  const hasNonAWEreports = nonAWEreports.length > 0;
-
+  const reportsFromB64 = createDiagnosticReportsWithB64Notes(diagnosticReports, observations);
+  const noReportsPlaceholder = `<table><tbody><tr><td>No reports found</td></tr></tbody></table>`;
   return `
     <div id="reports" class="section">
       <div class="section-title">
         <h3 id="reports" title="reports">&#x276F; Reports</h3>
         <a href="#mr-header">&#x25B2; Back to Top</a>
       </div>
       <div class="section-content">
-        ${
-          hasNonAWEreports
-            ? nonAWEreports
-            : `<table><tbody><tr><td>No reports found</td></tr></tbody></table>`
-        }
+        ${combineStrings(nonAWEreports, reportsFromB64, noReportsPlaceholder)}
       </div>
     </div>
   `;
 }
 
+function combineStrings(
+  stringA: string | undefined,
+  stringB: string | undefined,
+  placeholder = "default"
+) {
+  if (stringA && stringA.length > 0 && stringB && stringB.length > 0) {
+    return stringA + stringB;
+  } else if (stringA && stringA.length > 0) {
+    return stringA;
+  } else if (stringB && stringB.length > 0) {
+    return stringB;
+  }
+  return placeholder;
+}
+
+function createDiagnosticReportsWithB64Notes(
+  diagnosticReports: DiagnosticReport[],
+  observations: Observation[]
+) {
+  const mappedObservations = mapResourceToId<Observation>(observations);
+
+  if (!diagnosticReports) {
+    return "";
+  }
+
+  const timedNotes = diagnosticReports
+    .sort((a, b) => {
+      const timeA = a.effectiveDateTime ?? a.effectivePeriod?.start;
+      const timeB = b.effectiveDateTime ?? b.effectivePeriod?.start;
+      return dayjs(timeA).isBefore(dayjs(timeB)) ? -1 : 1;
+    })
+    .map(report => {
+      let time = report.effectiveDateTime ?? report.effectivePeriod?.start;
+
+      const obsRefs = report.result?.flatMap(obsRef => obsRef.reference || []);
+      const notes =
+        obsRefs
+          ?.flatMap(obsRef => {
+            const obsRefId = obsRef?.split("/")[1];
+            if (!obsRefId) return;
+            const referencedObs = mappedObservations[obsRefId];
+            if (!time) {
+              time = referencedObs?.effectiveDateTime ?? referencedObs?.effectivePeriod?.start;
+            }
+            const valueString = referencedObs?.valueString ?? "";
+            const cleanedUp = removeJunk(valueString);
+            const obsExtensions = referencedObs?.extension;
+            const resString = containsB64(obsExtensions)
+              ? Buffer.from(cleanedUp, "base64").toString("utf-8")
+              : cleanedUp;
+
+            return cleanUp(resString);
```

Comment:
nit: const cleanedUp is the result of removeJunk but then we call a different function called cleanUp. lets make the naming more consistent?